src/physics/material/MaterialParams.cc

```diff
@@ -198,39 +210,57 @@ MaterialParams::extend_elcomponents(const MaterialInput& inp)
  */
 void MaterialParams::append_material_def(const MaterialInput& inp)
 {
-    REQUIRE(inp.number_density >= 0);
-    REQUIRE((inp.number_density == 0) == inp.elements.empty());
```

Comment:
Revert the requirement here to allow the correct `void` material but catch someone passing a zero-number-density material with 3 elements.